lib/katello/tasks/upgrades/3.11/update_puppet_repos.rake

```diff
@@ -9,7 +16,8 @@ namespace :katello do
             puts "Refreshing repository #{repo.label} (#{repo.id})"
             install_dist = repo.backend_service(SmartProxy.pulp_master).backend_data['distributors'].find { |dist| dist['distributor_type_id'] == 'puppet_install_distributor' }
             if install_dist
-              SmartProxy.pulp_master.pulp_api.resources.repository.delete_distributor(repo.pulp_id, install_dist['id'])
+              response = SmartProxy.pulp_master.pulp_api.resources.repository.delete_distributor(repo.pulp_id, install_dist['id'])
```

Comment:
This is an async task, so the deletion will occur in the background.  The result is that this line passes fine, but the refresh down below will hit a race condition where it does:

1.  fetch distributors from pulp
2.  for each distributor, update the distributor to the expected configuration, otherwise create it

if the distributor is deleted between 1 and 2, katello will try to update a distributor that no longer exists and a 404 will be thrown.